sdk/cosmos/azure-cosmos/test/test_analytical_ttl.py

```diff
@@ -0,0 +1,296 @@
+
+#   In order for the analytical ttl tests to work you have to enable Azure Synapse Link for your Azure Cosmos DB
+#   account.
+
+
+@pytest.mark.usefixtures("teardown")
+class Test_ttl_tests(unittest.TestCase):
+    """TTL Unit Tests.
+    """
+
+    host = test_config._test_config.host
+    masterKey = test_config._test_config.masterKey
+    connectionPolicy = test_config._test_config.connectionPolicy
+
+    def __AssertHTTPFailureWithStatus(self, status_code, func, *args, **kwargs):
+        """Assert HTTP failure with status.
+
+        :Parameters:
+            - `status_code`: int
+            - `func`: function
+        """
+        try:
+            func(*args, **kwargs)
+            self.assertFalse(True, 'function should fail.')
+        except exceptions.CosmosHttpResponseError as inst:
+            self.assertEqual(inst.status_code, status_code)
+
+    @classmethod
+    def setUpClass(cls):
+        if (cls.masterKey == '[YOUR_KEY_HERE]' or
+                cls.host == '[YOUR_ENDPOINT_HERE]'):
+            raise Exception(
+                    "You must specify your Azure Cosmos account values for "
+                    "'masterKey' and 'host' at the top of this class to run the "
+                    "tests.")
+        cls.client = cosmos_client.CosmosClient(cls.host, cls.masterKey, consistency_level="Session",
+                                                connection_policy=cls.connectionPolicy)
+        cls.created_db = cls.client.create_database_if_not_exists("TTL_tests_database" + str(uuid.uuid4()))
+
+    def test_collection_ttl_replace_values(self):
+        created_collection = self.created_db.create_container_if_not_exists(
+            id='test_ttl_values1' + str(uuid.uuid4()),
+            partition_key=PartitionKey(path='/id'),
+            analytical_storage_ttl=-1)
+        self.created_db.replace_container(created_collection, partition_key=PartitionKey(path='/id'),
+                                          analytical_storage_ttl=1000)
+        created_collection_properties = created_collection.read()
+        self.assertEqual(created_collection_properties['analyticalStorageTtl'], 1000)
+
+        self.created_db.delete_container(container=created_collection)
+
+    def test_collection_and_document_ttl_values(self):
+        ttl = 10
+        created_collection = self.created_db.create_container_if_not_exists(
+            id='test_ttl_values1' + str(uuid.uuid4()),
+            partition_key=PartitionKey(path='/id'),
+            analytical_storage_ttl=ttl)
+        created_collection_properties = created_collection.read()
+        self.assertEqual(created_collection_properties['analyticalStorageTtl'], ttl)
+
+        collection_id = 'test_ttl_values4' + str(uuid.uuid4())
+        ttl = -10
+
+        # -10 is an unsupported value for ttl. Valid values are -1 or a non-zero positive 32-bit integer value
+        self.__AssertHTTPFailureWithStatus(
+            StatusCodes.BAD_REQUEST,
```

Comment:
A generic "bad request" error is not a good experience - what's the actual error message here? Would it tell users why the failure occurred and how to fix it?